sdk/identity/identity/src/credentials/authorizationCodeCredential.ts

```diff
@@ -184,7 +184,12 @@ export class AuthorizationCodeCredential implements TokenCredential {
 
       this.lastTokenResponse = tokenResponse;
       logger.getToken.info(formatSuccess(scopes));
-      return (tokenResponse && tokenResponse.accessToken) || null;
+      const token = (tokenResponse && tokenResponse.accessToken) || null;
+
+      if (token === null) {
+        throw new CredentialUnavailable("Failed to retrieve a valid token");
+      }
+      return token;
```

Comment:
NIT: if `tokenResponse && tokenResponse.accessToken` could be `undefined`, no need to change it to `null`. You can just check for `undefined` directly:
```suggestion
      const token = tokenResponse && tokenResponse.accessToken;

      if (token === undefined) {
        throw new CredentialUnavailable("Failed to retrieve a valid token");
      }
      return token;
```